doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -3,68 +3,234 @@
 Syntax extensions and notation scopes
 =====================================
 
-In this chapter, we introduce advanced commands to modify the way Coq
-parses and prints objects, i.e. the translations between the concrete
-and internal representations of terms and commands.
-
-The main commands to provide custom symbolic notations for terms are
-:cmd:`Notation` and :cmd:`Infix`; they will be described in the
-:ref:`next section <Notations>`. There is also a
-variant of :cmd:`Notation` which does not modify the parser; this provides a
-form of :ref:`abbreviation <Abbreviations>`. It is
-sometimes expected that the same symbolic notation has different meanings in
-different contexts; to achieve this form of overloading, Coq offers a notion
-of :ref:`notation scopes <Scopes>`.
-The main command to provide custom notations for tactics is :cmd:`Tactic Notation`.
+While, it's in theory possible to write down every Gallina term
+explicitly, in the way they are understood by the kernel, for instance
+:n:`Nat.add (S O) (Nat.mul (S (S O)) (S (S O)))`, this would be very
+tedious and error-prone and take us away from our usual mathematical
+practice. To circumvent this, Coq offers notation mechanisms that
+reflect the eponym mathematical practice and allows for instance to
+write :n:`1 + 2 * 2` instead. Those mechanisms range from simple
+:ref:`Abbreviations` to full fledged :ref:`Notations` with user
+defined :ref:`syntaxes <ReservingNotations>`. Multiple interpretations
+can be given to the same notation in different contexts thanks to the
+:ref:`scope <Scopes>` mechanism. For instance :n:`(1 + 2 * 2)%N` could
+denote the above natural number expression while :n:`(1 + 2 * 2)%Z`
+could denote a similar expression with integers.
```

Comment:
You mention several mechanisms (abbreviations, notations, syntaxes, scopes) but don't (briefly) say what they do/how they are useful or (for most) give tiny examples, which is precisely what an introduction should do.  Hard to make much sense of "multiple _interpretations_ can be given to the same notation in different _contexts_" without knowing what the italicized terms mean.  The doc overuses "context" with multiple meanings; it isn't very informative.

In the introduction text I pushed yesterday, I was careful to minimize the introduction of new terms and to give concrete examples for each new item.